src/Exercism.Representers.CSharp/Normalization/NormalizeDictionaryInitialization.cs

```diff
@@ -0,0 +1,183 @@
+                {
+                    if (initializerOperation.Parent?.Type?.Name != "Dictionary")
+                    {
+                        if (initializerOperation.Parent?.Type?.Name == default)
+                        {
+                            Log.Error(
+                                $"{nameof(NormalizeDictionaryInitialization)}: unable to retrieve the type information for {nameof(initializerOperation)}");
+                        }
+
+                        return DefaultVisit();
+                        // presumably a list or some object or other
+                    }
+
+                    var initializerExtractionResults
+                        = node.Kind() switch
+                        {
+                            SyntaxKind.CollectionInitializerExpression
+                            => ExtractInitializersWithCollectionSyntax(initializerOperation),
+                            SyntaxKind.ObjectInitializerExpression
+                            => ExtractInitializersWithObjectSyntax(initializerOperation),
+                            _ 
+                            => throw new InvalidKindException{Kind = node.Kind()}
+                        };
+
+                    if (!initializerExtractionResults.Success)
+                    {
+                        return DefaultVisit();
+                    }
+
+                    var visitedInitializerSyntaxNodes = initializerExtractionResults.InitializerSyntaxNodes
+                        .Select(initializer => new KeyValuePair<SyntaxNode, SyntaxNode>(
+                            this.Visit(initializer.Key),
+                            this.Visit(initializer.Value))
+                        );
+
+                    var dictionaryAsText = BuildDictionaryAsText(visitedInitializerSyntaxNodes);
+                    var dictionarySyntaxTree = CSharpSyntaxTree.ParseText(dictionaryAsText);
+                    var replacementNode = dictionarySyntaxTree
+                        .GetRoot()
+                        .DescendantNodes()
+                        .FirstOrDefault(n => n is InitializerExpressionSyntax);
+                    if (replacementNode == default)
+                    {
+                        Log.Error(
+                            $"{nameof(NormalizeDictionaryInitialization)}: failed to find a {nameof(InitializerExpressionSyntax)} node in generated dictionary fragment");
+                        return DefaultVisit();
+                    }
+
+                    return replacementNode;
+                }
+            }
+            catch (InvalidKindException ike)
+            {
+                Log.Error(
+                    $"{nameof(NormalizeDictionaryInitialization)}: {nameof(InitializerExpressionSyntax)} found with unexpected Kind {ike.Kind}");
+            }
+            catch (Exception e)
+            {
+                Log.Error(e, $"{nameof(NormalizeDictionaryInitialization)}: unknown error");
+            }
```

Comment:
I'm wondering if in these cases we should just let it crash, as it is likely that we have a bug in our code then. Now the representation is created and we won't notice there is anything wrong.

Reply:
I think we would spot it because we would have a non-matching representation.  

Is there any benefit from not crashing?  If we receive a solution that is in good order except that there is a dictionary problem.  We can add the representation, albeit a bit broken, and have time to investigate and fix the problem while the student carries on merrily.  We then withdraw the representation and substitute a fixed one.  Are my assumptions about the workflow correct?

Reply:
> We can add the representation, albeit a bit broken, and have time to investigate and fix the problem while the student carries on merrily.

How would we find out that the representation is a bit broken? Would there be something in the representation that the mentor would have to spot?

Reply:
if you're happy to crash, let's crash

Reply:
This was also partly based on @iHiD's recommendation :)